src/flb_io.c

```diff
@@ -610,7 +623,7 @@ int flb_io_net_write(struct flb_upstream_conn *u_conn, const void *data,
     flb_trace("[io thread=%p] [net_write] trying %zd bytes",
               th, len);
 
-    if (u->flags & FLB_IO_TCP) {
+    if (!u_conn->tls_session) {
```

Comment:
I am not sure about this change, would you please check the behavior for different plugins that have FLB_IO_TCP and FLB_IO_OPT_TLS ?

Reply:
A bit context on this line of change. Previously it's using the flag to decide if the read/write should be sent through TCP or TLS after encryption. For TCP connections, there is no behavior change because anyway there is no tls_session established (net_io_tls_handshake is only called when FLB_IO_TLS).

For FLB_IO_TLS, this change will not introduce issue because anyway the following code is assuming tls_session is not null. This change simply gives it a chance to send TCP payload without TLS encryption before TLS session is established. It's needed in the proxy case where proxy connect must be sent before TLS session.